tests/e2e/improved_csi_idempotency.go

```diff
@@ -838,5 +839,6 @@ func extendVolumeWithServiceDown(serviceName string, namespace string, client cl
 // stopHostD is a function for waitGroup to run stop hostd parallelly
 func stopHostD(ctx context.Context, addr string, wg *sync.WaitGroup) {
 	defer wg.Done()
+	defer ginkgo.GinkgoRecover()
```

Comment:
The failure is coming from https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/422e2784663f2b22652e972fb25c14f27bee0b65/tests/e2e/util.go#L3437

gingko is panic'ing since the failure and assertion gets triggered in a go routine

we can add a retry here - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/422e2784663f2b22652e972fb25c14f27bee0b65/tests/e2e/util.go#L3437 or handle the error in a better way when the func is called in a goroutine

since the current fix will only allow next test to run but will not fix the tests which are failing I don't think it is complete, lets try to fix the underlying issue

Also, raise a bug on the team to revisit all the blocks where we have assertions in methods which are called via goroutines